examples/directchat.nim

```diff
@@ -33,56 +31,45 @@ const Help = """
   exit: closes the chat
 """
 
-type
-  CustomData = ref object
-    consoleFd: AsyncFD
-    serveFut: Future[void]
-
-  ChatProto = ref object of LPProtocol
-    customData*: CustomData
-    switch: Switch
-    transp: StreamTransport
-    conn: Connection
-    client: bool
-    connected: bool
-    started: bool
-
-proc id (p: ChatProto): string =
-  if not isNil(p.conn.peerInfo):
-    $p.conn.peerInfo.peerId
+type ChatProto = ref object of LPProtocol
+  switch: Switch          # a single entry point for dialing and listening to peer
+  transp: StreamTransport # transport streams between read & write file descriptor 
+  conn: Connection        # create and close read & write stream
+  connected: bool         # if the node is connected to another peer
+  started: bool           # if the node has started
+
+
+# copied from https://github.com/status-im/nim-beacon-chain/blob/0ed657e953740a92458f23033d47483ffa17ccb0/beacon_chain/eth2_network.nim#L109-L115 
```

Comment:
we don't need this comment here

Reply:
It would be kind of hard to change the name of start.nim since it's used throughout the tutorials and all the screenshots..

Reply:
ah, that's too bad...